kornia_io/image.py

```diff
@@ -0,0 +1,85 @@
+from enum import Enum
+
+import cv2  # for reader
+import numpy as np
+
+from torch import Tensor
+import kornia as K
+from kornia.color import rgb_to_grayscale, bgr_to_grayscale
+
+# TODO: make it flexible
+__image_reader__ = cv2.imread
+
+
+class ImageColor(Enum):
+    GRAY = 0
+    RGB = 1
+    BGR = 2
+
+
+class Image(Tensor):
+    @staticmethod 
+    def __new__(cls, data, color, *args, **kwargs): 
+        return super().__new__(cls, data, *args, **kwargs) 
+
+    def __init__(self, data: Tensor, color: ImageColor) -> None:
+        self._color = color
+
+    @property
+    def is_batch(self) -> bool:
+        return len(self.data.shape) > 3
+
+    @property
+    def channels(self) -> int:
+        return self.data.shape[-3]
+
+    @property
+    def height(self) -> int:
+        return self.data.shape[-2]
+
+    @property
+    def width(self) -> int:
+        return self.data.shape[-1]
+
+    @property
+    def color(self) -> ImageColor:
+        return self._color
+
+    @classmethod
+    def from_tensor(cls, data: Tensor, color: ImageColor = ImageColor.RGB) -> 'Image':
+        return cls(data, color)
+
+    @classmethod
+    def from_file(cls, file_path: str) -> 'Image':
+        data: np.ndarray = __image_reader__(file_path)
+        data_t: Tensor = K.utils.image_to_tensor(data)
+        data_t = K.color.bgr_to_rgb(data_t)
+        # TODO: discuss whether we return normalised
+        data_t = data_t.float() / 255.
```

Comment:
another question is if you are sure that the the loaded is always int in range 0-255, some packages already gives normalized floats

Reply:
possibly we could pass a flag or something to normalize ?